pandas/core/indexing.py

```diff
@@ -1238,6 +1250,8 @@ def _convert_to_indexer(self, key, axis: int):
                 key = list(key)
 
             if com.is_bool_indexer(key):
+                # TODO: in this case should we do a .take on the value here?
+                # test_loc_setitem_all_false_boolean_two_blocks
```

Comment:
@phofl could use some extra eyeballs when convenient.  I've got this down to one failing test where we have an all-False mask indexer.  we need to do something like

```
if value_is_arraylike_etc:
     value = value[inds]
```

here, but not sure if this is the right place to do that

Reply:
Hm no, I think we have to handle this case later.
Something like
```
df = DataFrame({"a": [1, 2], "b": [3, 4]})
indexer = Series([False, False], name="c")
df.loc[[], ["b"]] = 1
```

would not go through there but raises the same error currently because the indexer is empty too.

Reply:
If we do not have a test for this case, we should add one I think. This passes on master right now